CDCgov/prime-reportstream
# 9397: search for providers, facilities, senders via new submitters API

This PR adds the endpoint for fetching the submitters for a particular receiver

Test Steps:
1. If you don't have any data locally, submit some reports through the legacy pipeline (I have data sent to the `ignore.HL7` receiver
2. Run 
```sh
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries/submitters/search' \
--header 'Content-Type: application/json' \
--data '{
    "sort": {
        "direction": "DESC",
        "property": "type"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [
        {
            "value": "2023-05-21T00:00:00+00:00",
            "filterName": "UNTIL"
        }
    ]
}'
```
and test that adjusting the filter (and trying the `SINCE` filter) works as expected as well as pagination and sorting.

## Changes
- Adds a new endpoint for fetching submitters to a particular receiver
- Updates `ReportGraph` to fetch metadata based on the item lineage rather than just the report lineage

## Checklist

### Testing

### Process

## Linked Issues
- #9397
- #9403 
Will be used to render the following:
![Screenshot 2023-06-06 at 12 04 05 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/d7bc24c7-f1d0-4cde-b613-26ada01e2ceb)


## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
